File: todo-reservation-springboot/src/main/java/de/vkb/dojo/es/reservations/model/ref/ReservationReference.java
package de.vkb.dojo.es.reservations.model.ref;

import com.fasterxml.jackson.annotation.*;
import de.vkb.dojo.es.common.model.ref.Reference;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("reservation")
public class ReservationReference implements Reference {
    private final String id;

    @JsonCreator
    public ReservationReference(
            @JsonProperty("id") String id
    ) {
        this.id = id;
    }

    @JsonGetter
    public String getId() {
        return id;
    }

    @Override
    public String getPath() {
        return "/reservations/" + id;
    }
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/reservations/model/ref/ReservationReference.kt
package de.vkb.dojo.es.reservations.model.ref

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonTypeName
import de.vkb.dojo.es.common.model.ref.Reference

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("reservation")
data class ReservationReference(
    override val id: String
): Reference {
    override val path: String = "/reservations/${id}"
}
